libs/pages/settings/src/lib/ui/page-organization-billing-summary/show-usage-modal/show-usage-modal.tsx

```diff
@@ -0,0 +1,179 @@
+import { type Organization } from 'qovery-typescript-axios'
+import { Controller, useFormContext } from 'react-hook-form'
+import { useOrganization } from '@qovery/domains/organizations/feature'
+import { InputSelect, InputText, ModalCrud } from '@qovery/shared/ui'
+
+export interface ShowUsageModalProps {
+  organizationId: string
+  renewalAt?: string | null
+  onSubmit: () => void
+  onClose: () => void
+  loading?: boolean
+}
+
+export class ReportPeriod {
+  from: string
+  to: string
+  option: ReportPeriodOption
+
+  constructor(from: string, to: string, option: ReportPeriodOption) {
+    this.from = from
+    this.to = to
+    this.option = option
+  }
+}
+
+export class ReportPeriodOption {
+  label: string
+  value: string
+
+  constructor(label: string, value: string) {
+    this.label = label
+    this.value = value
+  }
+}
+
+export function getReportPeriods(organization: Organization, orgRenewalAt: string | null | undefined): ReportPeriod[] {
+  const date = new Date()
+  date.setMonth(date.getMonth(), new Date(orgRenewalAt ? orgRenewalAt : organization.created_at).getDate())
+  date.setHours(0, 0, 0, 0)
+
+  const currentDateStr = new Intl.DateTimeFormat('en-US', {
+    dateStyle: 'medium',
+  }).format(date)
+
+  const reportPeriods = [
+    new ReportPeriod(
+      date.toISOString(),
+      new Date().toISOString(),
+      new ReportPeriodOption(`${currentDateStr} to Now`, 'current_month')
+    ),
+  ]
+
+  if (orgRenewalAt) {
+    const orgCreatedAtDate = new Date(organization.created_at)
+
+    // get last 12 months from renewalAt - 1 month if the month date is greater than org.created_at
+    for (let i = 1; i <= 12; i++) {
+      const renewalAtDate = new Date(orgRenewalAt)
+      const date = new Date()
+      date.setMonth(date.getMonth() - i, renewalAtDate.getDate())
+      date.setHours(0, 0, 0, 0)
+
+      if (date.getTime() < orgCreatedAtDate.getTime()) {
+        break
+      }
+
+      const nextMonthDate = new Date(date)
+      nextMonthDate.setMonth(nextMonthDate.getMonth() + 1)
+
+      // date formatting to Month, Day, Year
+      const dateToFormatx = new Intl.DateTimeFormat('en-US', {
+        dateStyle: 'medium',
+      }).format(new Date(date))
+
+      const nextMonthDateToFormat = new Intl.DateTimeFormat('en-US', {
+        dateStyle: 'medium',
+      }).format(new Date(nextMonthDate))
+
+      reportPeriods.push(
+        new ReportPeriod(
+          date.toISOString(),
+          nextMonthDate.toISOString(),
+          new ReportPeriodOption(`${dateToFormatx} to ${nextMonthDateToFormat}`, date.toISOString())
+        )
+      )
+    }
+  }
+
+  return reportPeriods
+}
+
+export function ShowUsageModal({ organizationId, renewalAt, onSubmit, onClose, loading }: ShowUsageModalProps) {
+  const { control } = useFormContext()
+
+  const { data: organization } = useOrganization({ organizationId })
+  const reportPeriods = getReportPeriods(organization as Organization, renewalAt)
```

Comment:
Normally it's not necessary to have an `as Organization` it's preferable to allow undefined on our function and return empty array if necessary